java/src/main/java/scenarios/crewcreation/CharacterCreation.java

```diff
@@ -0,0 +1,250 @@
+        } else if (meanScore == 9) {
+            System.out.println("I must say, you're exceptionally average.");
+            System.out.println("You're like a ruler for the rest of the universe to measure themselves against...");
+        } else if (meanScore >= 7) {
+            System.out.println("You're a bit below average, but I guess you've never let that stop you before...");
+        } else if (meanScore >= 4) {
+            System.out.println("These stats are quite low... life must be a bit of a struggle for you, huh?");
+        } else {
+            System.out.println("Wow... I don't think I've ever seen anything quite so terrible before.");
+            System.out.println("Just so you know, there's no shame in restarting the game.");
+            System.out.println("I probably would, if I were you...");
+        }
+
+        // TODO: (A) Have player assign their ability scores
+
+        Map<String, Integer> assignedAbilityScores = assignAbilityScores(unassignedAbilityScores);
+
+        //return null;
+        /* TODO: (A)
+        The above line is here so you can run the dice tests, otherwise the compiler would complain about using values
+        that haven't been set. Delete the line above and uncomment the line below when you get the dice tests passing
+        and you're ready to start adding the functionality for the player to set their ability scores using the values
+        that they rolled.
+        */
+        return new Captain(
+                name,
+                assignedAbilityScores.get(PHYSICAL),
+                assignedAbilityScores.get(AGILITY),
+                assignedAbilityScores.get(BLASTER),
+                assignedAbilityScores.get(PILOTING),
+                assignedAbilityScores.get(REPAIR),
+                assignedAbilityScores.get(CHARISMA)
+        );
+    }
+
+    private static Map<String, Integer> assignAbilityScores(List<Integer> scoresRolled) {
+        int end = 0;
+        Map<String, Integer> abilityScoreMap = new LinkedHashMap<>();
+
+        while (end == 0) {
+            abilityScoreMap.put(PHYSICAL, null);
+            abilityScoreMap.put(AGILITY, null);
+            abilityScoreMap.put(BLASTER, null);
+            abilityScoreMap.put(PILOTING, null);
+            abilityScoreMap.put(REPAIR, null);
+            abilityScoreMap.put(CHARISMA, null);
+
+            List<Integer> availableScores = new ArrayList<>(scoresRolled);
+            for (Map.Entry<String, Integer> ability : abilityScoreMap.entrySet()) {
+                String abilityName = ability.getKey();
+                System.out.println("");
+                System.out.println("Which score would you like to assign to your [" + abilityName + "] ability?");
+                System.out.println("Enter the number corresponding to the value you wish to use...");
+
+                // TODO: Depending on how Lanterna is set up, we should potentially extract the following code into a
```

Comment:
It's up to you, but if you want you can move the following code (including the TODO) to it's own method, something like `private static int selectScore(List<Integer> availableScores)`, in order to improve readability and make it easier to unit test later on, if we decide to test it (really need to see how this input related code will be handled after switching to Lanterna).